### PR TITLE
ext: redis client

### PR DESCRIPTION
**Pull request by tomaszgy**
* added extension for redis client

Signed-off-by: Tomasz <tomaszgy@gmail.com>

### CODE REVIEW

**Comment by jmartinm** [comment]
This looks :+1: to me. @kaplun @jalavik wdyt?

File: inspirehep/ext/redis/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (nikpap) [inline inspirehep/ext/redis/__init__.py]: I believe this can be removed nowadays.

Comment (jalavik) [inline inspirehep/ext/redis/__init__.py]: Unless something has changed, it might still be fine actually. We have it everywhere nowadays.

Comment (jalavik) [inline inspirehep/ext/redis/__init__.py]: Ah you mean the whole thing, yeah, right that can be removed. I thought you meant the encoding :dancer:

**Comment by jalavik** [comment]
Yeah :shipit:

File: inspirehep/ext/redis/__init__.py (hunk 1 above)
Comment (tomaszgy) [inline inspirehep/ext/redis/__init__.py]: So the __init__.py file is unnecessary?

**Comment by kaplun** [comment]
What is exactly this? Is it a way to be able to just use redis? Shouldn't this be on invenio_ext instead?

Why `__init__.py` is optional?

File: inspirehep/ext/redis/__init__.py (hunk 1 above)
Comment (jalavik) [inline inspirehep/ext/redis/__init__.py]: Sorry, it was not clear with our comments. We were discussing the license header. The file can just be empty.

File: inspirehep/ext/redis/client.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# or submit itself to any jurisdiction.
+
+
+redis = None
+
+
+def setup_app(app):
+
+    from werkzeug.contrib.cache import RedisCache
+    global redis
```
Comment (kaplun) [inline inspirehep/ext/redis/client.py]: Is this correct the correct pattern? Shouldn't this based context locals? http://werkzeug.pocoo.org/docs/0.11/local/

**Comment by jalavik** [comment]
@kaplun To connect to a Redis instance without the imposed timeout given by `invenio_ext.cache`. Who knows what will happen with `invenio_ext` in Invenio 3, so we figured to add it here for now and then later use the Invenio 3 way (if it exposes it's own thing).

**Comment by kaplun** [comment]
You mean the fact that on Invenio cache things expires by default? I see.

File: inspirehep/ext/redis/__init__.py (hunk 1 above)
Comment (tomaszgy) [inline inspirehep/ext/redis/__init__.py]: OK, I see.

**Comment by jalavik** [comment]
@kaplun yeah, exactly.

File: inspirehep/ext/redis/client.py (hunk 2 above)
Comment (tomaszgy) [inline inspirehep/ext/redis/client.py]: Well, I am not sure though I followed the pattern from https://github.com/inspirehep/invenio-ext/blob/master/invenio_ext/es.py#L81

Comment (kaplun) [inline inspirehep/ext/redis/client.py]: OK, just had the proof from https://gitter.im/inveniosoftware/invenio that that pattern was not correct. (it has been removed in 3).

I think we should really use context locals, here (and in invenio-ext as well). `globals` are evil (e.g. if use multithreading).

**Comment by jmartinm** [comment]
@tomaszgy Here you have another example in the new `invenio-search` of how to use `LocalProxy`

https://github.com/inveniosoftware/invenio-search/blob/master/invenio_search/proxy.py#L27-L37

You can use it as an example to avoid the `global` use.

**Comment by jmartinm** [comment]
I like the fact that you can import Redis with:

```python
from inspirehep.ext.redis import redis
```

I am wondering if it is too much boilerplate, for example, registering another extension https://github.com/inspirehep/inspire-next/pull/603/files#diff-56eeba39c106acfdd01c4ce0609975b6R39

Do you think it can be simplified just with `setup_app` and having the `LocalProxy` ?

**Comment by tomaszgy** [comment]
Well I agree it's a bit boilerplate although you can see it in many invenio_exts. I guess it is kind of a pattern. I tried to find a simpler way of doing that but I haven't really found anything that'd look better.

**Comment by jmartinm** [comment]
Alright, if there are no more comments I think we can :shipit:

**Comment by jalavik** [comment]
:+1:

**Pull request merged**